src/platform/forms-system/src/js/fields/AutosuggestField.jsx

```diff
@@ -118,37 +129,48 @@ export default class AutosuggestField extends React.Component {
     return _.set('widget', 'autosuggest', suggestion);
   };
 
+  getItemFromInput = (inputValue, suggestions, uiOptions) => {
+    let item = { widget: 'autosuggest', label: inputValue };
+    // once the input is long enough, check for exactly matching strings so that we don't
+    // force a user to click on an item when they've typed an exact match of a label
+    if (inputValue && inputValue.length > 3) {
+      const matchingItem = suggestions.find(
+        suggestion => suggestion.label === inputValue,
+      );
+      if (matchingItem) {
+        item = this.getFormData(matchingItem);
+      }
+    }
+
+    const { freeInput, inputTransformers } = uiOptions;
+
+    const inputToSave =
+      inputTransformers &&
+      Array.isArray(inputTransformers) &&
+      inputTransformers.length
+        ? inputTransformers.reduce(
+            (userInput, transformer) => transformer(userInput),
+            inputValue,
+          )
+        : inputValue;
+
+    return freeInput || this.useEnum ? inputToSave : item;
```

Comment:
is `freeInput` just a boolean? should `getItemFromInput` always return a string?

Reply:
It is a boolean, but I don't think this function always returns a string. It returns an string or an object, depending on if we're letting people type freely (and not select a value) or using an enum list from the schema, or just querying an api for the list.